Upstream-Status: Inappropriate [configuration]

Use ?= for Makefile variables to use poky environment variables.
Original method to sed s/CC =/CC ?=/ is not applicable anymore
because of the Makefile changes.

against 3.86

06/28/2010 - qhe

also AUXDIR stays in $(LIBDIR)/syslinux rather than $(DATADIR)/syslinux

07/08/2010 - qhe

updated for 4.03

11/16/2010 - Qing He <qing.he@intel.com>

---
diff --git a/MCONFIG b/MCONFIG
index e9c16d3..4d49f33 100644
--- a/MCONFIG
+++ b/MCONFIG
@@ -18,13 +18,13 @@
 MAKEFLAGS += -r
 MAKE      += -r
 
-BINDIR   = /usr/bin
-SBINDIR  = /sbin
-LIBDIR   = /usr/lib
-DATADIR  = /usr/share
-AUXDIR   = $(DATADIR)/syslinux
-MANDIR	 = /usr/man
-INCDIR   = /usr/include
+BINDIR   ?= /usr/bin
+SBINDIR  ?= /sbin
+LIBDIR   ?= /usr/lib
+DATADIR  ?= /usr/share
+AUXDIR   = $(LIBDIR)/syslinux
+MANDIR	 ?= /usr/man
+INCDIR   ?= /usr/include
 TFTPBOOT = /tftpboot
 COM32DIR = $(AUXDIR)/com32
 
@@ -38,18 +38,18 @@ PERL	 = perl
 
 CHMOD	 = chmod
 
-CC	 = gcc
+CC	 ?= gcc
 gcc_ok   = $(shell tmpf=gcc_ok.$$$$.tmp; \
 		   if $(CC) $(GCCOPT) $(1) -c $(topdir)/dummy.c \
 			-o $$tmpf 2>/dev/null ; \
 		   then echo '$(1)'; else echo '$(2)'; fi; \
 		   rm -f $$tmpf)
 
-LD	 = ld
-OBJDUMP	 = objdump
-OBJCOPY  = objcopy
-AR       = ar
-NM       = nm
-RANLIB   = ranlib
+LD	 ?= ld
+OBJDUMP	 ?= objdump
+OBJCOPY  ?= objcopy
+AR       ?= ar
+NM       ?= nm
+RANLIB   ?= ranlib
 GZIPPROG = gzip
 PNGTOPNM = pngtopnm
 MCOPY    = mcopy
